pydantic/_internal/_std_types_schema.py

```diff
@@ -90,47 +91,32 @@ def to_enum(input_value: Any, /) -> Enum:
         try:
             return enum_type(input_value)
         except ValueError:
-            # The type: ignore on the next line is to ignore the requirement of LiteralString
-            raise PydanticCustomError('enum', f'Input should be {expected}', {'expected': expected})  # type: ignore
+            raise PydanticCustomError('enum', 'Input should be {expected}', {'expected': expected})
 
-    strict_python_schema = core_schema.is_instance_schema(enum_type)
-
-    to_enum_validator = core_schema.no_info_plain_validator_function(to_enum)
     if issubclass(enum_type, int):
         # this handles `IntEnum`, and also `Foobar(int, Enum)`
-        updates['type'] = 'integer'
-        lax = core_schema.chain_schema([core_schema.int_schema(), to_enum_validator])
-        # Disallow float from JSON due to strict mode
-        strict = core_schema.json_or_python_schema(
-            json_schema=core_schema.no_info_after_validator_function(to_enum, core_schema.int_schema()),
-            python_schema=strict_python_schema,
-        )
+        js_updates['type'] = 'integer'
+        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.int_schema())
```

Comment:
I've replaced lots of use of `core_schema.chain_schema` with `core_schema.no_info_after_validator_function` which should be more efficient, and is more concise.